packages/netsuite-adapter/src/deletion_calculator.ts

```diff
@@ -0,0 +1,152 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ElemID, InstanceElement } from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { NetsuiteQuery, ObjectID } from './query'
+import { getCustomRecordCounters, getCustomRecordTypeInstances } from './changes_detector/changes_detectors/custom_records'
+import NetsuiteClient from './client/client'
+import { ElemServiceID, LazyElementsSourceIndexes } from './elements_source_index/types'
+import { SCRIPT_ID } from './constants'
+
+const getCurrentScriptIds = async (elementsSourceIndex: LazyElementsSourceIndexes): Promise<ElemServiceID[]> => {
+  const serviceIdRecords = (await elementsSourceIndex.getIndexes()).serviceIdRecordsIndex
+  return Object.values(serviceIdRecords)
+    .filter(({ elemID }) => elemID.nestingLevel === 1 && elemID.name === SCRIPT_ID)
+    .map(({ elemID, serviceID }) => ({ elemID: elemID.createParentID(), serviceID }))
+}
+
+const calculateDeletedInstances = (
+  currentInstances: ElemServiceID[],
+  serviceInstancesIds: ObjectID[],
+): ElemID[] => {
+  if (currentInstances.length === 0) {
+    return []
+  }
+
+  const serviceIdsGroupByTypes = _(serviceInstancesIds)
+    .groupBy(objectId => objectId.type)
+    .mapValues(objectIds => objectIds.map(({ instanceId }) => instanceId))
+    .value()
+
+  return currentInstances
+    .filter(item => !serviceIdsGroupByTypes[item.elemID.typeName]?.includes(item.serviceID))
```

Comment:
custom record type needs a special treat:
```suggestion
    .filter(item => !serviceIdsGroupByTypes[item.elemID.idType === 'type' ? CUSTOM_RECORD_TYPE : item.elemID.typeName]?.includes(item.serviceID))
```

Reply:
where? not sure that I saw it...